azure-kusto-ingest/source/utils.ts

```diff
@@ -8,12 +10,45 @@ export const fileToStream = (file: FileDescriptor | string): StreamDescriptor =>
     return new StreamDescriptor(streamFs, fileDescriptor.sourceId, compressionType);
 }
 
-export const sleep = (ms: number) => {
+export const sleep = (ms: number): Promise<void> => {
     return new Promise((resolve) => {
         setTimeout(resolve, ms);
     });
 }
 
-export const getRandomSleep = (baseMs: number) => {
+export const getRandomSleep = (baseMs: number): number => {
     return baseMs + Math.floor(Math.random() * 1000);
-  }
+}
+
+export const mergeStreams = (...streams: Readable[]) : Readable => {
+    let pass = new PassThrough()
+    let waiting = streams.length
+    for (const stream of streams) {
+        pass = stream.pipe(pass, { end: false })
+        stream.once('end', () => --waiting === 0 && pass.emit('end'))
+    }
+    return pass
+}
+
+
+export const tryStreamToSizedArray = async (stream: Readable, maxBufferSize: number): Promise<Buffer | Readable> => {
```

Comment:
rename to tryStreamToArray? I think that becuase you have maxBufferSize parameter you don't need to add the "Sized"

Reply:
Done, but I still think the name is kind of unclear and not describes the function well, I just can't think of another one

Reply:
I agree, this is a confusing function in general...
Maybe it should return some tuple? So we won't have to check instanceof later?

Reply:
I don't know, because then we'll be checking for nulls.
This typing is stronger